sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/ConnectionString.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.communication.common;
+
+import java.util.Objects;
+
+/**
+ * Parse and store Connection String values
+ */
+public class ConnectionString {
```

Comment:
Make this class `final` and also add `@Immutable` annotation. Also, the user doesn't directly create this type, so, it can be moved to implementation package.